clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -210,8 +209,7 @@ public int partitionLeaderEpoch() {
     private CloseableIterator<Record> compressedIterator() {
         ByteBuffer buffer = this.buffer.duplicate();
         buffer.position(RECORDS_OFFSET);
-        final DataInputStream stream = new DataInputStream(compressionType().wrapForInput(
-                new ByteBufferInputStream(buffer), magic()));
+        final DataInputStream stream = new DataInputStream(compressionType().wrapForInput(buffer, magic()));
```

Comment:
I'd prefer to do that in a follow-up PR in trunk, since I'd also like to backport this to 0.10.2.x